third_party/nvfuser/csrc/scheduler/normalization_utils.cpp

```diff
@@ -568,26 +568,57 @@ bool hasSharedInput(
 bool hasSharedConsumer(
     const std::vector<TensorView*>& inner_reduction_tvs,
     const std::vector<TensorView*>& outer_reduction_tvs) {
-  // check inner reduction tvs
-  auto contvs = ir_utils::consumerTvsOf(inner_reduction_tvs[0]);
-  std::set<TensorView*> contvs_set(contvs.begin(), contvs.end());
-  for (int i = 1; i < (int)inner_reduction_tvs.size(); i++) {
-    for (auto tv : ir_utils::consumerTvsOf(inner_reduction_tvs[i])) {
-      if (contvs_set.find(tv) != contvs_set.end()) {
-        return true;
-      } else {
-        contvs_set.emplace(tv);
+  auto getAllConsumerTvsOf = [](TensorView* tv0) {
```

Comment:
There's several routines for this type of analyses in iter_visitor.h. I think in this case you could just use `DependencyCheck::getAllOutputsOf`.

Reply:
changed to `DependencyCheck::getAllOutputsOf`